metis-technical-metadata-generation/src/main/java/eu/europeana/metis/technical/metadata/generation/utilities/MediaExtractorForFile.java

```diff
@@ -155,6 +194,18 @@ private void submitResource(final FileStatus fileStatus, final Set<Integer> nonR
     });
   }
 
+  private void thumbnailUpload(String resourceUrl) {
+    final TechnicalMetadataWrapper technicalMetadataWrapper = mongoDao
+        .getTechnicalMetadataWrapper(resourceUrl);
+    if (CollectionUtils.isEmpty(technicalMetadataWrapper.getThumbnailWrappers())) {
+      LOGGER.info("Resource does not have thumbnails: {}", resourceUrl);
+    } else {
+      storeThumbnailsToS3(amazonS3Client, s3Bucket,
```

Comment:
Since this method does not throw an exception, the result of a problem is that the thumbnails are removed without them having been added to S3. Is this desired behavior? Even in case of connectivity issues, for instance?

Reply:
Fixed. It will not remove the thumbnails if any of the thumbnails failed upload.